### PR TITLE
Features/game nights controller

### PR DESCRIPTION
**Pull request by npauzenga**
Hey @Enriikke! 

Here are specs for the `GameNights` controller. I think these (and all my controller specs) could be DRY'd up quite a bit. I've already got a `Helpers` module that I'm using for `authenticate` and maybe it's doable to make things like `params` and the interactor inputs/contexts more generic.

OR maybe I could just move a lot of these `let`s into a module. It just seems like things get hard to read when you've got the `params`  AND the `interactor_input` AND the `interactor_context` AND the Interactor stub on every single CRUD action.

### CODE REVIEW

**Comment by enriikke** [comment]
Awesome work @npauzenga!! :shipit: 

I see what you mean about DRYing the specs a bit. What I tend to do is define the variables used on all specs as a global `let`. For example:

``` ruby
let(:params) do
  {
    id:                 something_id,
    game_night: game_night
  }
end

let(:something_id) { ... }
let(:game_night) { ... } 
```

The cool thing there is that you might only need to update a small portion of `params` per spec. That cleans up everything a bit. Generalizing also helps where you can. I would say that personally prefer DRYing up from within the spec itself when possible.

**Comment by npauzenga** [comment]
~~Hey @Enriikke here are the passing specs on this guy. How do you feel about my handling of the `#destroy` action? Would you prefer I render a message if the action is successful?~~

Sorry, this was redundant. Your explanation in the previous PR is great. Just take a glance at the whole thing :smile:

**Comment by npauzenga** [comment]
Hey @Enriikke! Please have a look at this new implementation. I've generalized all of the CRUD interactors so we can just re-use the same 4-5 in every controller.

This'll close #74

File: app/controllers/game_nights_controller.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
+class GameNightsController < AuthenticationController
+  def create
+    result = CreateModel.call(active_record_class: GameNight,
```
Comment (npauzenga) [inline app/controllers/game_nights_controller.rb:3]: @Enriikke can you think of a better way to extract the class? This is a little inelegant :smile_cat:

**Comment by npauzenga** [comment]
@Enriikke Bah! I think the Interactor refactor is a no-go.

I didn't consider my join tables. When we create a `Group` we don't want just `Group.create(...)`, we want `user.groups.create(...)` to get the associations along with it. And then we need to change the user's role to reflect their new `moderator` status.

And that's similar but different from creating a `GameNight` where we want `group.game_nights.create(...)` and then find the `current_user` to add their role as `organizer`.

And that's entirely different from creating a User :smile: 

I think that, in order to keep the generalized `CreateModel` interactor it would need like 7 really confounding arguments. Maybe there's a way using callbacks on the Group/GameNight models but that seems hairy.

Dang it. I'll wait for you to have a look at all this but I think the best bet is rolling back to 408342d3d4f4ccf94aaf593ef2643236b0b29164

**Comment by npauzenga** [comment]
Leaving this open for reference but this will close in favor of #99

**Pull request closed without merging**